### DISCUSSION TITLE
Disable DEBUG logging - only for rocketry

### DISCUSSION THREAD

**Discussion by psetzermann** [Miksus/rocketry#209] [Q&A] [answered]
Hi, 

is there a possiblity to disable the logging (level=DEBUG) specifically for rocketry.
The rest of my application should still log (DEBUG) messages but I don't want to see 
"[DEBUG - rocketry.scheduler -            run_cycle()]  Beginning cycle with 2 tasks..." 
every 100 milliseconds. 

Cheers
Paul

**Comment by psetzermann** [comment]
Anyone? @Miksus :)

**Comment by Jypear** [comment, marked as answer]
@psetzermann 
This shouldn't be too tricky to do, it would just require a bit of manipulation of the logging module.
Specifically you would just need to change the logging level on the rocketry logger before running your scheduled tasks.

Example:
```
from rocketry import Rocketry
import logging

app = Rocketry()

logging.basicConfig(level=logging.DEBUG, format="%(asctime)s - %(levelname)s -  %(message)s")
logging.getLogger("rocketry.scheduler").setLevel(logging.INFO)

@app.task("every 3 seconds")
def do_task():
    print("Im inside the task")
    logging.debug("This is my log inside the task")

app.run()
```

output:
```
2023-08-23 12:52:00,802 - DEBUG -  Using selector: KqueueSelector
2023-08-23 12:52:00,803 - INFO -  Starting up...
2023-08-23 12:52:00,804 - INFO -  Startup complete.
2023-08-23 12:52:00,907 - INFO -  Task 'do_task' status: 'run'
Im inside the task
2023-08-23 12:52:00,908 - DEBUG -  This is my log inside the task
2023-08-23 12:52:00,909 - INFO -  Task 'do_task' status: 'success'
```

The above ignores the spammy log message you're talking about. If you just wanted to ignore that single log message without changing the logging level of the rocketry.scheduler logger, I imagine there could probably be some way to use a logging filter.

Cheers,
Josh

**Reply by psetzermann** [reply]
Thats exactly what I needed. "rocketry.scheduler" was the key. 
Thanks & Cheers, 
Paul
